dbms/src/Flash/Pipeline/Schedule/Tasks/OneTimeNotifyFuture.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Flash/Pipeline/Schedule/Tasks/NotifyFuture.h>
+#include <Flash/Pipeline/Schedule/Tasks/PipeConditionVariable.h>
+
+#include <mutex>
+
+namespace DB
+{
+class OneTimeNotifyFuture : public NotifyFuture
```

Comment:
can you add some comments to explain why it is named as `OneTimeNotifyFuture`

Reply:
ok, comment added.